addons/sale_coupon/models/sale_order.py

```diff
@@ -48,8 +48,9 @@ def action_confirm(self):
         valid_coupon_ids.write({'state': 'new', 'partner_id': self.partner_id})
         (self.generated_coupon_ids - valid_coupon_ids).write({'state': 'cancel', 'partner_id': self.partner_id})
         self.applied_coupon_ids.write({'state': 'used'})
+        res = super(SaleOrder, self).action_confirm()
```

Comment:
```suggestion
        res = super().action_confirm()
```

Reply:
Or another solution would be to override `_action_confirm` instead of `action_confirm` :thinking: